lib/writeShieldConfig.js

```diff
@@ -78,8 +63,12 @@ function createRoles(configPath, roles) {
   });
 }
 
-module.exports = function writeShieldConfig(options) {
+module.exports = function writeShieldConfig(options, version) {
   return function (configPath) {
+    // append the shield/xpack directory to configPath
+    var usersConfig = semver.satisfies(version, '5.x || 3.x') ? 'xpack' : 'shield';
+    configPath = path.join(configPath, usersConfig);
```

Comment:
You should create a new variable here instead of just appending the `usersConfig` to this. When this function finally returns `configPath` it's now pointing to the wrong directory (which other functions depend on being set to the actual `configPath`)